tests/beacon/state_machines/test_demo.py

```diff
@@ -1,17 +1,74 @@
+import pytest
+
+
+from eth._utils import bls as bls
 from eth.beacon.db.chain import BeaconChainDB
-from eth.beacon.types.blocks import BaseBeaconBlock
-from eth.beacon.types.states import BeaconState
+from eth.beacon.enums import (
+    SignatureDomain,
+)
+from eth.beacon.helpers import (
+    get_beacon_proposer_index,
+)
+from eth.beacon.state_machines.forks.serenity.blocks import (
+    SerenityBeaconBlock,
+)
+from eth.beacon.types.proposal_signed_data import ProposalSignedData
 
 
-def test_demo(base_db, sample_beacon_block_params, sample_beacon_state_params, fixture_sm_class):
+@pytest.mark.parametrize(
+    (
+        'num_validators,'
+        'epoch_length,'
+        'min_attestation_inclusion_delay,'
+        'target_committee_size,'
+        'shard_count'
+    ),
+    [
+        (10, 2, 1, 2, 2)
+    ]
+)
+def test_demo(base_db,
+              sample_beacon_block_params,
+              genesis_state,
+              fixture_sm_class,
+              config,
+              privkeys,
+              pubkeys):
     chaindb = BeaconChainDB(base_db)
-    state = BeaconState(**sample_beacon_state_params)
-    block = BaseBeaconBlock(**sample_beacon_block_params).copy(
-        slot=state.slot + 2,
+    state = genesis_state
+    block = SerenityBeaconBlock(**sample_beacon_block_params).copy(
+        slot=state.slot + 1,
+    )
+
+    # Sign block
+    beacon_proposer_index = get_beacon_proposer_index(
+        state.copy(
```

Comment:
You can just use `state` here. The current `state.slot` will give the look ahead on the proposer slot as long as it's not past an epoch transition.

Reply:
Ahhh! We need to fix bug https://github.com/ethereum/eth2.0-specs/issues/352 first so that we can use `state`. 😭

Reply:
Leave a TODO comment for now:
```
# TODO: use `state` when the bug of `get_shard_committees_at_slot` is fixed.
```